apps/sms/views/shared/test/unit/activity_handler_test.js

```diff
@@ -943,6 +940,24 @@ suite('ActivityHandler', function() {
         );
       }).then(done,done);
     });
+
+    test('existing thread with specified body, navigate to composer',
```

Comment:
Agree, it depends on the side we look at it :) My main points were:
* Make sure that we redirect immediately to view without waiting for the thread search operation to complete (should be noticeable by user anyway though);
* Make test a bit less confusing since we reject/resolve something that doesn't have any effect.

Both are minor points, and "black box" thing you're talking about totally makes sense to me as well - so I'm okay with your intention :)